### PR TITLE
switch dunfell to tegra boot tools

### PR DESCRIPTION
**Pull request by tirankas**
These are a collection of commits pulled together to switch the dunfell 32.5.0 branch to use Tegra boot tools instead of the redundant boot.

### CODE REVIEW

**Comment by madisongh** [comment]
@tirankas could you add a Signed-off-by line to the commit message, as described in our [contributing guidelines](https://github.com/OE4T/meta-tegra/blob/master/CONTRIBUTING.md)? I can take care of the necessary newline fixups I mentioned in [the discussion thread](https://github.com/OE4T/meta-tegra/discussions/798#discussioncomment-1357644).

Thanks!

**Comment by tirankas** [comment]
Hi @madisongh 
I committed the spaces & new lines fixes (removing tabs, fixing indentation and added new lines).
I just now saw your request for signed-off. How do I do that given the commit is made...

**Comment by tirankas** [comment]
Hi @madisongh 
Is it OK now?

Thanks

File: layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-bsp/tegra-bup-payload/tegra-bup-payload_%.bbappend [hunk 1]
```diff
@@ -0,0 +1,2 @@
+RDEPENDS_${PN}_remove = "tegra-redundant-boot"
+RDEPENDS_${PN}_append = " tegra-boot-tools-updater"
```
Comment (ichergui) [inline layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-bsp/tegra-bup-payload/tegra-bup-payload_%.bbappend]: Could you please add new line please ?

File: layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-bsp/u-boot/libubootenv_%.bbappend [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+# Also stash an extra copy of the fw_env.config file in the
+# rootfs so we can compare it to the currently-installed one
+# in a state script, to see if we need to take steps to save
+# and restore u-boot environment variables during a bootloader
+# update.
+do_install_append_tegra() {
+    install -d ${D}${datadir}/u-boot
+    install -m 0644 ${WORKDIR}/fw_env.config ${D}${datadir}/u-boot/
+}
+FILES_${PN}_append_tegra = " ${datadir}/u-boot"
```
Comment (ichergui) [inline layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-bsp/u-boot/libubootenv_%.bbappend]: Same as above

File: layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-core/initrdscripts/tegra-minimal-init/platform-preboot-cboot.sh [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    if [ -n "$mayberoot" ]; then
+	rootdev="$mayberoot"
+    else
+	rootdev="/dev/mmcblk0p1"
+    fi
+fi
+
+if which bootcountcheck >/dev/null 2>&1; then
+    bootcountcheck
+fi
```
Comment (ichergui) [inline layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-core/initrdscripts/tegra-minimal-init/platform-preboot-cboot.sh]: same as above

File: layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-mender/tegra-state-scripts/tegra-state-scripts_%.bbappend [hunk 4]
```diff
@@ -1 +1,20 @@
...
+    cp ${S}/redundant-boot-commit-check-script-uboot ${MENDER_STATE_SCRIPTS_DIR}/ArtifactCommit_Enter_80_bl-check-update
+}
+
+copy_other_scripts() {
+    :
+}
+
+do_compile_tegra210() {
+    cp ${S}/redundant-boot-install-script-uboot ${MENDER_STATE_SCRIPTS_DIR}/ArtifactInstall_Leave_80_bl-update
+}
```
Comment (ichergui) [inline layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-mender/tegra-state-scripts/tegra-state-scripts_%.bbappend]: Same as above

**Review by ichergui — CHANGES_REQUESTED** [review]

File: layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-mender/tegra-state-scripts/tegra-state-scripts_%.bbappend (hunk 4 above)
Comment (tirankas) [reply layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-mender/tegra-state-scripts/tegra-state-scripts_%.bbappend]: Hi @ichergui 
I already resolved all these comment with the latest commit done an hour ago.
Thanks

Comment (ichergui) [reply layers/meta-tegra-support/dynamic-layers/meta-mender-tegra/recipes-mender/tegra-state-scripts/tegra-state-scripts_%.bbappend]: Sorry tirankas
I didn't refresh the PR. 
Thanks for the update

**Comment by ichergui** [comment]
> Hi @madisongh
> Is it OK now?
> 
> Thanks

Could you please add `signed-off-by` in this [commit](https://github.com/OE4T/tegra-demo-distro/pull/154/commits/6a582f6df6866ca1d27cc51217bdc33d51862ec9) and this [one](https://github.com/OE4T/tegra-demo-distro/pull/154/commits/c5af2611c40399a198ec4fbf7300302aa586333f) ?

Thanks

**Comment by madisongh** [comment]
@ichergui  I'll take care of fixing that up as part of the merge.  There are a few other tweaks needed as well, based on my testing.

**Comment by ichergui** [comment]
> @ichergui I'll take care of fixing that up as part of the merge. There are a few other tweaks needed as well, based on my testing.

@madisongh 
I can do the test as well, I have my `Jetson Xavier NX` here.
Please let me know if I can help to validate this PR

**Comment by tirankas** [comment]
Hi @madisongh 
I am not able to sign the other two commits, it gives an error:
error: gpg failed to sign the data
fatal: failed to write commit object

Not sure where this comes from.
I did sign one commit...

Command used is: git commit --amend -S -c 6a582f6df6866ca1d27cc51217bdc33d51862ec9

**Comment by madisongh** [comment]
@tirankas GPG signing isn't required.  All I needed was the Signed-off-by line, which you've already added.

Thanks

**Comment by tirankas** [comment]
Great, thanks!
I will verify again on the NX Xavier when my build completed. Will probably need to verify again on the nano. Will post when I am done.

**Comment by madisongh** [comment]
I've tested on Nano (SD and eMMC), TX2, and Xavier NX (eMMC), updating dunfell-32.4.3 -> 32.5.x -> 32.6.1.  Found one issue in tegra-boot-tools (specific to t210 platforms), and have fixed that.

Nano (SD): 32.4.3->32.5.x fails as expected; 32.5.x -> 32.6.1 OK
All updates for all other succeed as expected.

**Comment by madisongh** [comment]
I've manually merged this.

Thanks again, @tirankas !

**Comment by dwalkes** [comment]
Thanks @madisongh and @tirankas !  I see this made it into the `dunfell` branch as well in https://github.com/OE4T/tegra-demo-distro/commit/dbb64a206e2cc38b2f4207116ad169d1bdf23f7c
So based on previous discussion in https://github.com/OE4T/meta-tegra/discussions/798#discussioncomment-1351843 I think we are now on `tegra-boot-tools` for all branches dunfell-l4t-r32.4.3 or later in tegra-demo-distro.

**Pull request closed without merging**